pkg/cloud-provider/cloudapi/azure/azure_crd_helpers.go

```diff
@@ -103,8 +114,10 @@ func computeInstanceToInternalVirtualMachineObject(instance *virtualMachineTable
 	} else {
 		state = runtimev1alpha1.Unknown
 	}
-	return utils.GenerateInternalVirtualMachineObject(crdName, strings.ToLower(cloudName), strings.ToLower(cloudID), strings.ToLower(region),
-		namespace, strings.ToLower(cloudNetworkID), cloudNetworkShortID, state, tags, networkInterfaces, providerType, account)
+
+	return utils.GenerateInternalVirtualMachineObject(crdName, strings.ToLower(cloudName), strings.ToLower(cloudID),
```

Comment:
we are passing too many arguments here.. can you create new type or consider creating the object here and fill renaming stuff not available here in utils?

Reply:
Moved VPC, VM object creation functions from crd_generator.go to cloud specific files. 
Renamed crd_generator.go to helpers.go